types/fqdn.pp

```diff
@@ -1,2 +1,2 @@
 # @summary Validate a Fully Qualified Domain Name
-type Stdlib::Fqdn = Pattern[/\A(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])\z/]
+type Stdlib::Fqdn = Pattern[/\A((([[:alnum:]]|[[:alnum:]][[:alnum:]-]*[[:alnum:]])\.)*[[:alpha:]]+)\z/]
```

Comment:
i dont think we should change to the posix expressions as they also include unicode and although DNS has IDN's many places where you would enter a domain name (including many DNS daemons) expect the name to be in the ASCII punycode format .  At the very least this should be in a separate PR so its clearly highlighted and marked as a change in behaviour  (possibly even backwards incompatible)

Reply:
Oh :hankey:! I was 100% sure that `[a-z]` included things like `é` and that I was changing nothing here!  Thank you for spotting this, I will revert to the list of allowed chars!

Reply:
Reverted!